### DISCUSSION TITLE
New Markdown Element

### DISCUSSION THREAD

**Discussion by chenyuntc** [zauberzeug/nicegui#3343] [Ideas / Feature Requests]
Current markdown element appearance is subpar compared to Streamlit's. Are there plans to replace python-markdown2 with more modern alternatives like [Quasar UI QMarkdown](https://quasarframework.github.io/quasar-ui-qmarkdown/demo) or [vue3-markdown](https://www.npmjs.com/package/vue3-markdown) or [md-editor-v3](https://imzbf.github.io/md-editor-v3/en-US/index) for better visual consistency with NiceGUI?

**Comment by falkoschindler** [comment]
Thanks for the suggestion, @chenyuntc!
Is there anything in particular that you're missing in `ui.markdown`? Or can you show a visual comparison with other libraries?
It is certainly possible to add another or replace the existing markdown element - if there is a good reason to do so.
Alternatively we could provide themes to add CSS/Tailwind for an alternative look.

**Comment by chenyuntc** [comment]
Thanks for the your reply!

I generally feel streamlit markdown style looks more visual apealing, here are a few comparison (I use markdown sample from https://markdown-it.github.io/)

Takes code for example (in nicegui I use `extras=['code_friendly','fenced-code-blocks'])`
Streamlit looks more visual appealing, furthermore, it provides a convenient float button to copy all code inside a code block.

![image](https://github.com/user-attachments/assets/32f89e88-89e3-494c-aedc-55d634140762) ![image](https://github.com/user-attachments/assets/f007b329-b5d6-4baf-a96f-16f75127d59c)


**and the font size looks a bit odd in python-markdown**

![image](https://github.com/user-attachments/assets/45cdcceb-b6b4-4013-924c-d7b7a66b84a7)
![image](https://github.com/user-attachments/assets/c10e82b6-61a5-4af4-a208-192029048eea)
